dali/sasha/saxref.cpp

```diff
@@ -355,25 +372,57 @@ struct cDirDesc
         return fn;
     }
 
-    bool isMisplaced(unsigned partNum, unsigned numParts, const SocketEndpoint &ep, IGroup &grp)
+    bool isMisplaced(unsigned partNum, unsigned numParts, const SocketEndpoint &ep, IGroup &grp, const char *name, const char *scope, unsigned stripeNum, unsigned numStripedDevices)
     {
-        // MORE: Add better check for misplaced in Containerized
-        // If plane being scanned is host based (i.e. not locally mounted), misplaced could still make sense
         if (isContainerized())
-            return false;
+        {
+            // Remove dir-per-part from scope for hashing filename and comparing to part number
+            unsigned dirPerPartNum = 0;
+            const char *lastScope = nullptr;
+            StringBuffer scopeBuf(scope);
+            if (scopeBuf.length())
+            {
+                lastScope = strrchr(scope, ':');
+                if (!lastScope)
+                    lastScope = scope;
+                else
+                    lastScope++;
+                dirPerPartNum = getDirPerPartNum(lastScope);
+            }
+            if (numStripedDevices>1)
+            {
+                if (dirPerPartNum !=0 && dirPerPartNum <= numParts)
+                    scopeBuf.setLength(lastScope - scope);
+                else
+                    scopeBuf.append("::");
+                // Remove file mask from filename for hashing
+                const char *ext = strrchr(name,'.');
+                if (!ext)
+                    ext = name + strlen(name);
+                unsigned lfnHash = getFilenameHash(scopeBuf.append(ext-name,name).str());
+                if ((stripeNum>numStripedDevices)||(stripeNum<1))
```

Comment:
minor: could move this code earlier in the if() so that the use of lfnHash is nearer to its calculation.